services/status/src/main.rs

```diff
@@ -626,6 +627,29 @@ fn wrapped_main() -> ! {
     #[cfg(any(feature="precursor", feature="renode"))]
     llio.clear_wakeup_alarm().unwrap(); // this is here to clear any wake-up alarms that were set by a prior coldboot command
 
+    // get the last status
+    let must_sleep = early_settings.early_sleep().unwrap();
+    
+    // reset it for good measure
+    early_settings.set_early_sleep(false).unwrap();
```

Comment:
This seems to trigger a patch on every reboot. Since the sector is fixed, this means that sector gets rewritten on every reboot.

I guess this is probably actually not so bad, since "full cold" reboots are theoretically pretty rare, but it might be worth it to first check if it's actually necessary before calling it?

The spinor patching routine is reasonably clever in that it also diffs the sector before patching it to reduce wear and tear but I think it's better to avoid having calls that aggravate flash wearout if we can avoid it.

Reply:
So instead of having the caller check that, the server does it internally before every patch write… I can work on that. Thanks!

Reply:
@bunnie refactored as you suggested.

Also noticed hosted mode is broken because mapping pages panics, I'll have to fix that later down the line.